### PR TITLE
Fix issue reading from empty queue.

### PR DESCRIPTION
**Pull request by ztiffany**
Fixes ofi-cray/libfabric-cray#192.

Signed-off-by: Zach Tiffany ztiffany@cray.com

### CODE REVIEW

**Comment by lanl-ompi** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.open-mpi.org/job/edison_libfabric_pr/105/
Test PASSed.

File: prov/gni/src/gnix_cq.c [hunk 1]
```diff
@@ -495,7 +492,7 @@ static ssize_t gnix_cq_readfrom(struct fid_cq *cq, void *buf, size_t count,
 
 	fastlock_release(&cq_priv->lock);
 
-	return read_count;
+	return read_count ?: -FI_EAGAIN;
```
Comment (hppritcha) [inline prov/gni/src/gnix_cq.c:495]: just curious is the triad def such that if there is nothing before the : will it return the
read_count in this case?  I'd prefer to have it more explicit and have
`return read_count ? read_count : -FI_EAGAIN:`

Comment (ztiffany) [inline prov/gni/src/gnix_cq.c:495]: Yes, it's equivalent to:

return read_count ? read_count : -FI_EAGAIN:

**Comment by hppritcha** [comment]
I'm good with merging this.

**Pull request merged**